d2/src/main/java/com/linkedin/d2/balancer/properties/UriPropertiesMerger.java

```diff
@@ -48,6 +55,10 @@ public UriProperties merge(String propertyName, Collection<UriProperties> proper
       }
     }
 
+    if (maxVersion == -1)
+    {
+      _log.warn("Merged Uri properties has invalid version -1. It should be > -1.");
+    }
```

Comment:
Should these checks be outside the `UriPropertiesMerger` code and somewhere in the dual read comparison code, since this class isn't really concerned and/or responsible about such validations?

It might also result in a lot of logs where not really required.

Reply:
the one in [ZooKeeperEphemeralStore.java](https://github.com/linkedin/rest.li/pull/958/files#diff-77b01f0ad40699c361791ff221b474e25279aba3277103a4d7abe7ad5d1a2d3b) logs about the invalid data and property name. Here is where the merge-version logic happens, so it's responsible for the merged final version.
The dual read monitor is not responsible for this validation, and the property there is generic. But this version issue only applies to uris. 
I've add a warn log to the xds flow uri property as well.

> It might also result in a lot of logs where not really required.
Any of this warn message is important since it means some invalid zk (and future Kafka) data and need our attention to fix.